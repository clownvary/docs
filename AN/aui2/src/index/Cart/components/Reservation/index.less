@import '~shared/style/variables/index.less';
/* stylelint-disable */

.reservation {
  line-height: 24px;

  .align-right {
    text-align: right;
  }

  .reservation-title {
    font-size: 20px;
    font-weight: @semibold;
  }

  .reservation-header,
  .reservation-list {
    .afx-col {
      font-size: 14px;
    }
  }

  .reservation-header {
    padding: 30px 10px 12px;
    color: @color-black-lighter;
    border-bottom: solid 2px @color-gray-lighter;
  }

  .reservation-list {
    border-bottom: solid 2px @color-gray-lighter;

    .reservation-item {
      .summary {
        padding: 10px 10px 0;
        font-size: 16px;
        line-height: 1.5;

        .name {
          font-weight: @semibold;
        }
      }

      .action-group {
        padding: 6px 10px;
        font-size: 14px;
        line-height: 24px;
        color: @color-blue;

        .action {
          margin-right: 20px;
          cursor: pointer;

          &:hover {
            color: @color-blue-darker;
          }

          .icon {
            padding: 5px;
            color: @color-blue;
          }
        }
      }

      .detail-list {
        padding-left: 20px;
        background-color: @color-blue-lighter;

        .detail-item {
          padding: 10px;
          vertical-align: middle;
          border-top: dotted 1px @color-gray-lighter;

          .name {
            font-size: 16px;
          }

          &__name {
            width: 325px;
            overflow: hidden;
            text-overflow: ellipsis;
            word-break: break-word;
            word-wrap: break-word;
            white-space: nowrap;
          }

          &__center {
            flex: 3;
            overflow: hidden;
            text-overflow: ellipsis;
            word-break: break-word;
            word-wrap: break-word;
            white-space: nowrap;

            > span:first-child {
              margin-right: 8px;
              font-weight: bolder;
            }
          }

          &__amount {
            flex: 1;
          }
        }
      }
    }
  }

  .modal-box {
    width: 350px !important;
    height: 300px !important;
  }

  .reservation-footer {
    padding: 10px 10px 0;
    font-size: 14px;
    text-align: right;

    .sub-fee {
      .fee-name {
        display: inline-block;
        padding-right: 28px;
        font-weight: @semibold;
      }

      .fee-amount {
        display: inline-block;
      }
    }

    .total {
      padding-top: 10px;
      font-size: 20px;
      font-weight: @semibold;
      line-height: 150%;

      .text-color-strong {
        padding-left: 20px;
      }
    }
  }
}

.cart-empty {
  padding-top: 50px;
  margin-top: 100px;
  text-align: center;

  .icon-cart-m {
    font-size: 65px;
    cursor: default;
    color: #d9e1e4;
  }

  .message {
    padding-top: 20px;
    font-size: 24px;
    line-height: 100%;
    text-align: center;
    color: #748288;
  }

  .link {
    margin-top: 10px;
    color: @color-blue;

    .icon {
      margin-right: 5px;
    }
  }
}
