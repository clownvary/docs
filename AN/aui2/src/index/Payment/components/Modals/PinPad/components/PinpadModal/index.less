.pinpad-modal {
  .modal-close {
    display: none;
  }

  .modal-body {
    padding: 30px;
    border-radius: 0 0 12px 12px;
  }

  #pinpad-applet-container {
    height: 0;
  }
}
