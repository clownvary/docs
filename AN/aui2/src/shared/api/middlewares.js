import PromiseMiddleware from './promiseMiddleware';
import RouteMiddleware from './routeMiddleware';
import ThunkMiddleware from './thunkMiddleware';
import API from './API';

export default [
  new ThunkMiddleware(),
  new RouteMiddleware(),
  new PromiseMiddleware(new API())
];
