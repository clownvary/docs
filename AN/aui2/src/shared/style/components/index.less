// Common
@import './page.less';
@import './panel.less';

// Components
@import './authority.less';
@import './datepicker.less';
@import './modal.less';
@import './checkbox.less';
@import './radio.less';
