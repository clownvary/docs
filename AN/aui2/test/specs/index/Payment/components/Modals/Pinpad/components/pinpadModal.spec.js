import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { fromJS } from 'immutable';
import configureStore from 'redux-mock-store';
import middlewares from 'shared/api/middlewares';
import { PinpadModal } from 'index/Payment/components/Modals/PinPad/components/PinpadModal';

jest.mock('react-base-ui/lib/components/Modal', () => 'Modal');
jest.mock('index/Payment/components/Modals/PinPad/components/Pinpad/mutiplePays', () => 'MutiplePays');
jest.mock('index/Payment/components/Modals/PinPad/components/Pinpad/singlePay', () => 'SinglePay');
jest.mock('index/Payment/components/Modals/PinPad/components/Pinpad/applet', () => 'Applet');

describe('index/Payment/components/Modals/PinPad/components/PinpadModal', () => {

  let store;

  beforeEach(() => {
    store = configureStore(middlewares)({});
  });
  const actions = {
    communicatingWithDevice: jest.fn(),
    updateErrorMessage: jest.fn(),
    updateCurrentPayIndex: jest.fn(),
    updatePromptMessage: jest.fn(),
    getAPDServerStatus: jest.fn(),
    processAPDResponse: jest.fn(),
    updateSuccessMessage: jest.fn(),
    processResolveAction: jest.fn(),
    processRejectAction: jest.fn(),
    clearError: jest.fn(),
    addError: jest.fn(),
    showFailActionBtns: jest.fn(),
    updateModalTitle: jest.fn(),
    showModal: jest.fn(),
    setApdInterfaceApplet: jest.fn()
  };
  const resetActions = () => Object.keys(actions).forEach(fn => actions[fn].mockClear());

  const props = {
    pinpad: {
      payment: fromJS({
        receiptHeaderId: 1234,
        agentId: 0,
        isRefund: false,
        currentPayIndex: 0,
        promptMessage: "",
        isNewCard: false,
        cardHolderInfo: {
          cardholderZip: null,
          cardholderAddr: null
        },
        pays: [
          {
            ams_account_id: "APD AccountID",
            payment_type_id: 4,
            index: 0,
            payName: "Debit Card",
            amount: "370.00",
            apd_input_type: 1,
            paymentTypeId: 4,
            errorMessage: 'unknown error',
            hasError: true
          }
        ],
        communicating: false,
        debitCardId: 4,
        companyId: 0,
        customerId: 1,
        apdAppletInfo: {
          device_type_id: 1,
          terminal_location_address: "937 Enterprise Dr",
          device_id: 999888,
          terminal_location_city: "Sacramento",
          terminal_location_state: "CA",
          terminal_location_zip: "95825",
          card_acceptor_name: "YMCA OF METROPOLITAN LOS ANGELES",
          merchant_descriptor: "ActiveNetTrainer",
          merchant_user: "AMS34VbVMSCRealDivId",
          merchant_password: "AMS34VbVMSCRealDivId",
          ams_url: "mstest.active.com",
          ams_port: 443,
          ams_timeout: 100,
          terminal_number: "007",
          usb_port_name: "COM5",
          pinpad_data_entry_timeout: 20,
          pinpad_swipe_card_timeout: 50,
          pinpad_pin_entry_timeout: 40,
          pinpad_device_timeout: 300,
          avs_full_addr: true,
          avs_zip: false,
          write_to_log: "Yes",
          applet_name: "APDInterfaceApplet",
          applets: "codebase=/device-libs/"
        },
        errorMessage: "",
        apdPaymentInfo: {
          isPayerCashCustomer: false,
          hasImmediatePaymentDue: false,
          canCancelApdPayment: true,
          key: 1,
          committedReceiptId: 22916,
          receiptPaymentId: 20423,
          rno: 1,
          receiptNumber: 1000778.005,
          orderId: "Receipt ID 22916",
          orderDescriptor: "YMCA OF METROPOLITAN LOS ANGELES"
        }
      }),
      pinpadModal: fromJS({
        title: "Processing Transaction",
        shown: false,
        apdInterfaceApplet: {}
      }),
      pinpadFail: fromJS({
        shown: false
      })
    }
  };
  const setup = (store, props) => mount(<PinpadModal store={store} {...props} {...actions} />);

  beforeEach(() => {
    resetActions();
  });

  it('component should render correctly when hasn`t shown modal', () => {
    const component = setup(store, props);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should render correctly when has show modal and has two payments by pinpad', () => {
    const payment = props.pinpad.payment.update('pays', pays => pays.push([{
      ams_account_id: "APD AccountID",
      payment_type_id: 3,
      index: 0,
      payName: "Credit Card",
      amount: "123.00",
      apd_input_type: 1,
      paymentTypeId: 3
    }]));
    const pinpadModal = props.pinpad.pinpadModal.set('shown', true);
    const newProps = {
      pinpad: {
        ...props.pinpad,
        payment,
        pinpadModal
      }
    };
    const component = setup(store, newProps);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should render correctly when has shown modal and only has one payment', () => {
    const pinpadModal = props.pinpad.pinpadModal.set('shown', true);
    const newProps = {
      pinpad: {
        ...props.pinpad,
        pinpadModal
      }
    };
    const component = setup(store, newProps);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should render correctly when has shown modal and has no payments', () => {
    const payment = props.pinpad.payment.update('pays', pays => pays.splice(0));
    const pinpadModal = props.pinpad.pinpadModal.set('shown', true);
    const newProps = {
      pinpad: {
        ...props.pinpad,
        payment,
        pinpadModal
      }
    };
    const component = setup(store, newProps);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should work fine', () => {
    const component = setup(store, props);

    expect(component).toBeTruthy();
    let apdInfo = component.instance().initAndStartAPD();
    apdInfo.methods.updateMessage('message');
    expect(actions.updatePromptMessage).toHaveBeenCalled();
    apdInfo.methods.updateErrMsgAndEnableFailBtns('error message');
    expect(actions.updateErrorMessage).toHaveBeenCalled();
    expect(actions.communicatingWithDevice).toHaveBeenCalledTimes(1);
    expect(actions.showFailActionBtns).toHaveBeenCalled();
    apdInfo.methods.newCardFromAPD({});
    expect(actions.communicatingWithDevice).toHaveBeenCalledTimes(2);
    expect(actions.updateSuccessMessage).toHaveBeenCalled();
    apdInfo.methods.clearPinpad();
    expect(actions.clearError).toHaveBeenCalled();
    expect(actions.addError).toHaveBeenCalled();
    expect(actions.showModal).toHaveBeenCalled();
    expect(actions.updateModalTitle).toHaveBeenCalled();
    expect(actions.showFailActionBtns).toHaveBeenCalled();
    expect(actions.processRejectAction).toHaveBeenCalled();

    __STATIC__ = false;
    const payment = props.pinpad.payment.set('currentPayIndex', 1)
      .set('isNewCard', true)
      .update('pays', pays => pays.push([{
        ams_account_id: "APD AccountID",
        payment_type_id: 3,
        index: 0,
        payName: "Credit Card",
        amount: "123.00",
        apd_input_type: 1,
        paymentTypeId: 3
      }]));
    component.setProps({
      pinpad: {
        ...props.pinpad,
        payment,
        pinpadModal: fromJS({
          shown: true
        })
      }
    });
    resetActions();
    apdInfo = component.instance().initAndStartAPD();
    apdInfo.methods.newCardFromAPD({});
    expect(actions.communicatingWithDevice).toHaveBeenCalled();
    expect(actions.processResolveAction).toHaveBeenCalled();

    __STATIC__ = true;
    const payment2 = payment.set('currentPayIndex', 0);
    const newProps = {
      pinpad: {
        ...props.pinpad,
        payment: payment2,
        pinpadModal: fromJS({
          shown: true
        })
      }
    };
    component.setProps(newProps);
    resetActions();
    apdInfo = component.instance().initAndStartAPD();
    apdInfo.methods.checkAPDServerStatus();
    expect(actions.getAPDServerStatus).toHaveBeenCalled();
    apdInfo.methods.processAPDResponse({});
    expect(actions.processAPDResponse).toHaveBeenCalled();
  });

  it('component should work fine when __STATIC__ is false', () => {
    __STATIC__ = false;
    const component = setup(store, props);
    expect(component).toBeTruthy();
    __STATIC__ = true;
    component.unmount();
  });
});
