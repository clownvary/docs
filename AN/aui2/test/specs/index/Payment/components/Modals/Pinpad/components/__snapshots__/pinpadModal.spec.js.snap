// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/Payment/components/Modals/PinPad/components/PinpadModal component should render correctly when has show modal and has two payments by pinpad 1`] = `
<PinpadModal
  addError={[Function]}
  clearError={[Function]}
  communicatingWithDevice={[Function]}
  getAPDServerStatus={[Function]}
  pinpad={
    Object {
      "payment": Immutable.Map {
        receiptHeaderId: 1234,
        agentId: 0,
        isRefund: false,
        currentPayIndex: 0,
        promptMessage: "",
        isNewCard: false,
        cardHolderInfo: Immutable.Map {
            cardholderZip: null,
            cardholderAddr: null,
        },
        pays: Immutable.List [
            Immutable.Map {
                ams_account_id: "APD AccountID",
                payment_type_id: 4,
                hasError: true,
                index: 0,
                payName: "Debit Card",
                amount: "370.00",
                apd_input_type: 1,
                paymentTypeId: 4,
                errorMessage: "unknown error",
            },
            Array [
                Object {
                  "amount": "123.00",
                  "ams_account_id": "APD AccountID",
                  "apd_input_type": 1,
                  "index": 0,
                  "payName": "Credit Card",
                  "paymentTypeId": 3,
                  "payment_type_id": 3,
                },
              ],
        ],
        communicating: false,
        debitCardId: 4,
        companyId: 0,
        customerId: 1,
        apdAppletInfo: Immutable.Map {
            pinpad_data_entry_timeout: 20,
            avs_zip: false,
            avs_full_addr: true,
            write_to_log: "Yes",
            merchant_user: "AMS34VbVMSCRealDivId",
            device_id: 999888,
            merchant_descriptor: "ActiveNetTrainer",
            pinpad_pin_entry_timeout: 40,
            ams_timeout: 100,
            terminal_location_state: "CA",
            terminal_number: "007",
            terminal_location_address: "937 Enterprise Dr",
            usb_port_name: "COM5",
            card_acceptor_name: "YMCA OF METROPOLITAN LOS ANGELES",
            applet_name: "APDInterfaceApplet",
            terminal_location_city: "Sacramento",
            merchant_password: "AMS34VbVMSCRealDivId",
            ams_url: "mstest.active.com",
            device_type_id: 1,
            pinpad_swipe_card_timeout: 50,
            applets: "codebase=/device-libs/",
            pinpad_device_timeout: 300,
            ams_port: 443,
            terminal_location_zip: "95825",
        },
        errorMessage: "",
        apdPaymentInfo: Immutable.Map {
            receiptNumber: 1000778.005,
            hasImmediatePaymentDue: false,
            orderId: "Receipt ID 22916",
            receiptPaymentId: 20423,
            rno: 1,
            committedReceiptId: 22916,
            canCancelApdPayment: true,
            isPayerCashCustomer: false,
            orderDescriptor: "YMCA OF METROPOLITAN LOS ANGELES",
            key: 1,
        },
    },
      "pinpadFail": Immutable.Map {
        shown: false,
    },
      "pinpadModal": Immutable.Map {
        title: "Processing Transaction",
        shown: true,
        apdInterfaceApplet: Immutable.Map {
        },
    },
    }
  }
  processAPDResponse={[Function]}
  processRejectAction={[Function]}
  processResolveAction={[Function]}
  setApdInterfaceApplet={[Function]}
  showFailActionBtns={[Function]}
  showModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  updateCurrentPayIndex={[Function]}
  updateErrorMessage={[Function]}
  updateModalTitle={[Function]}
  updatePromptMessage={[Function]}
  updateSuccessMessage={[Function]}
>
  <Modal
    className="pinpad-modal"
    shown={true}
    title="Processing Transaction"
  >
    <div
      className="modal-body"
    >
      <MutiplePays
        pinpad={
          Object {
            "payment": Immutable.Map {
              receiptHeaderId: 1234,
              agentId: 0,
              isRefund: false,
              currentPayIndex: 0,
              promptMessage: "",
              isNewCard: false,
              cardHolderInfo: Immutable.Map {
                  cardholderZip: null,
                  cardholderAddr: null,
              },
              pays: Immutable.List [
                  Immutable.Map {
                      ams_account_id: "APD AccountID",
                      payment_type_id: 4,
                      hasError: true,
                      index: 0,
                      payName: "Debit Card",
                      amount: "370.00",
                      apd_input_type: 1,
                      paymentTypeId: 4,
                      errorMessage: "unknown error",
                  },
                  Array [
                      Object {
                        "amount": "123.00",
                        "ams_account_id": "APD AccountID",
                        "apd_input_type": 1,
                        "index": 0,
                        "payName": "Credit Card",
                        "paymentTypeId": 3,
                        "payment_type_id": 3,
                      },
                    ],
              ],
              communicating: false,
              debitCardId: 4,
              companyId: 0,
              customerId: 1,
              apdAppletInfo: Immutable.Map {
                  pinpad_data_entry_timeout: 20,
                  avs_zip: false,
                  avs_full_addr: true,
                  write_to_log: "Yes",
                  merchant_user: "AMS34VbVMSCRealDivId",
                  device_id: 999888,
                  merchant_descriptor: "ActiveNetTrainer",
                  pinpad_pin_entry_timeout: 40,
                  ams_timeout: 100,
                  terminal_location_state: "CA",
                  terminal_number: "007",
                  terminal_location_address: "937 Enterprise Dr",
                  usb_port_name: "COM5",
                  card_acceptor_name: "YMCA OF METROPOLITAN LOS ANGELES",
                  applet_name: "APDInterfaceApplet",
                  terminal_location_city: "Sacramento",
                  merchant_password: "AMS34VbVMSCRealDivId",
                  ams_url: "mstest.active.com",
                  device_type_id: 1,
                  pinpad_swipe_card_timeout: 50,
                  applets: "codebase=/device-libs/",
                  pinpad_device_timeout: 300,
                  ams_port: 443,
                  terminal_location_zip: "95825",
              },
              errorMessage: "",
              apdPaymentInfo: Immutable.Map {
                  receiptNumber: 1000778.005,
                  hasImmediatePaymentDue: false,
                  orderId: "Receipt ID 22916",
                  receiptPaymentId: 20423,
                  rno: 1,
                  committedReceiptId: 22916,
                  canCancelApdPayment: true,
                  isPayerCashCustomer: false,
                  orderDescriptor: "YMCA OF METROPOLITAN LOS ANGELES",
                  key: 1,
              },
          },
            "pinpadFail": Immutable.Map {
              shown: false,
          },
            "pinpadModal": Immutable.Map {
              title: "Processing Transaction",
              shown: true,
              apdInterfaceApplet: Immutable.Map {
              },
          },
          }
        }
      />
      <div
        id="pinpad-applet-container"
      />
    </div>
  </Modal>
</PinpadModal>
`;

exports[`index/Payment/components/Modals/PinPad/components/PinpadModal component should render correctly when has shown modal and has no payments 1`] = `
<PinpadModal
  addError={[Function]}
  clearError={[Function]}
  communicatingWithDevice={[Function]}
  getAPDServerStatus={[Function]}
  pinpad={
    Object {
      "payment": Immutable.Map {
        receiptHeaderId: 1234,
        agentId: 0,
        isRefund: false,
        currentPayIndex: 0,
        promptMessage: "",
        isNewCard: false,
        cardHolderInfo: Immutable.Map {
            cardholderZip: null,
            cardholderAddr: null,
        },
        pays: Immutable.List [
        ],
        communicating: false,
        debitCardId: 4,
        companyId: 0,
        customerId: 1,
        apdAppletInfo: Immutable.Map {
            pinpad_data_entry_timeout: 20,
            avs_zip: false,
            avs_full_addr: true,
            write_to_log: "Yes",
            merchant_user: "AMS34VbVMSCRealDivId",
            device_id: 999888,
            merchant_descriptor: "ActiveNetTrainer",
            pinpad_pin_entry_timeout: 40,
            ams_timeout: 100,
            terminal_location_state: "CA",
            terminal_number: "007",
            terminal_location_address: "937 Enterprise Dr",
            usb_port_name: "COM5",
            card_acceptor_name: "YMCA OF METROPOLITAN LOS ANGELES",
            applet_name: "APDInterfaceApplet",
            terminal_location_city: "Sacramento",
            merchant_password: "AMS34VbVMSCRealDivId",
            ams_url: "mstest.active.com",
            device_type_id: 1,
            pinpad_swipe_card_timeout: 50,
            applets: "codebase=/device-libs/",
            pinpad_device_timeout: 300,
            ams_port: 443,
            terminal_location_zip: "95825",
        },
        errorMessage: "",
        apdPaymentInfo: Immutable.Map {
            receiptNumber: 1000778.005,
            hasImmediatePaymentDue: false,
            orderId: "Receipt ID 22916",
            receiptPaymentId: 20423,
            rno: 1,
            committedReceiptId: 22916,
            canCancelApdPayment: true,
            isPayerCashCustomer: false,
            orderDescriptor: "YMCA OF METROPOLITAN LOS ANGELES",
            key: 1,
        },
    },
      "pinpadFail": Immutable.Map {
        shown: false,
    },
      "pinpadModal": Immutable.Map {
        title: "Processing Transaction",
        shown: true,
        apdInterfaceApplet: Immutable.Map {
        },
    },
    }
  }
  processAPDResponse={[Function]}
  processRejectAction={[Function]}
  processResolveAction={[Function]}
  setApdInterfaceApplet={[Function]}
  showFailActionBtns={[Function]}
  showModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  updateCurrentPayIndex={[Function]}
  updateErrorMessage={[Function]}
  updateModalTitle={[Function]}
  updatePromptMessage={[Function]}
  updateSuccessMessage={[Function]}
>
  <Modal
    className="pinpad-modal"
    shown={true}
    title="Processing Transaction"
  >
    <div
      className="modal-body"
    >
      <div
        id="pinpad-applet-container"
      />
    </div>
  </Modal>
</PinpadModal>
`;

exports[`index/Payment/components/Modals/PinPad/components/PinpadModal component should render correctly when has shown modal and only has one payment 1`] = `
<PinpadModal
  addError={[Function]}
  clearError={[Function]}
  communicatingWithDevice={[Function]}
  getAPDServerStatus={[Function]}
  pinpad={
    Object {
      "payment": Immutable.Map {
        receiptHeaderId: 1234,
        agentId: 0,
        isRefund: false,
        currentPayIndex: 0,
        promptMessage: "",
        isNewCard: false,
        cardHolderInfo: Immutable.Map {
            cardholderZip: null,
            cardholderAddr: null,
        },
        pays: Immutable.List [
            Immutable.Map {
                ams_account_id: "APD AccountID",
                payment_type_id: 4,
                hasError: true,
                index: 0,
                payName: "Debit Card",
                amount: "370.00",
                apd_input_type: 1,
                paymentTypeId: 4,
                errorMessage: "unknown error",
            },
        ],
        communicating: false,
        debitCardId: 4,
        companyId: 0,
        customerId: 1,
        apdAppletInfo: Immutable.Map {
            pinpad_data_entry_timeout: 20,
            avs_zip: false,
            avs_full_addr: true,
            write_to_log: "Yes",
            merchant_user: "AMS34VbVMSCRealDivId",
            device_id: 999888,
            merchant_descriptor: "ActiveNetTrainer",
            pinpad_pin_entry_timeout: 40,
            ams_timeout: 100,
            terminal_location_state: "CA",
            terminal_number: "007",
            terminal_location_address: "937 Enterprise Dr",
            usb_port_name: "COM5",
            card_acceptor_name: "YMCA OF METROPOLITAN LOS ANGELES",
            applet_name: "APDInterfaceApplet",
            terminal_location_city: "Sacramento",
            merchant_password: "AMS34VbVMSCRealDivId",
            ams_url: "mstest.active.com",
            device_type_id: 1,
            pinpad_swipe_card_timeout: 50,
            applets: "codebase=/device-libs/",
            pinpad_device_timeout: 300,
            ams_port: 443,
            terminal_location_zip: "95825",
        },
        errorMessage: "",
        apdPaymentInfo: Immutable.Map {
            receiptNumber: 1000778.005,
            hasImmediatePaymentDue: false,
            orderId: "Receipt ID 22916",
            receiptPaymentId: 20423,
            rno: 1,
            committedReceiptId: 22916,
            canCancelApdPayment: true,
            isPayerCashCustomer: false,
            orderDescriptor: "YMCA OF METROPOLITAN LOS ANGELES",
            key: 1,
        },
    },
      "pinpadFail": Immutable.Map {
        shown: false,
    },
      "pinpadModal": Immutable.Map {
        title: "Processing Transaction",
        shown: true,
        apdInterfaceApplet: Immutable.Map {
        },
    },
    }
  }
  processAPDResponse={[Function]}
  processRejectAction={[Function]}
  processResolveAction={[Function]}
  setApdInterfaceApplet={[Function]}
  showFailActionBtns={[Function]}
  showModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  updateCurrentPayIndex={[Function]}
  updateErrorMessage={[Function]}
  updateModalTitle={[Function]}
  updatePromptMessage={[Function]}
  updateSuccessMessage={[Function]}
>
  <Modal
    className="pinpad-modal"
    shown={true}
    title="Processing Transaction"
  >
    <div
      className="modal-body"
    >
      <SinglePay
        pay={
          Object {
            "amount": "370.00",
            "ams_account_id": "APD AccountID",
            "apd_input_type": 1,
            "errorMessage": "unknown error",
            "hasError": true,
            "index": 0,
            "payName": "Debit Card",
            "paymentTypeId": 4,
            "payment_type_id": 4,
          }
        }
        pinpad={
          Object {
            "payment": Immutable.Map {
              receiptHeaderId: 1234,
              agentId: 0,
              isRefund: false,
              currentPayIndex: 0,
              promptMessage: "",
              isNewCard: false,
              cardHolderInfo: Immutable.Map {
                  cardholderZip: null,
                  cardholderAddr: null,
              },
              pays: Immutable.List [
                  Immutable.Map {
                      ams_account_id: "APD AccountID",
                      payment_type_id: 4,
                      hasError: true,
                      index: 0,
                      payName: "Debit Card",
                      amount: "370.00",
                      apd_input_type: 1,
                      paymentTypeId: 4,
                      errorMessage: "unknown error",
                  },
              ],
              communicating: false,
              debitCardId: 4,
              companyId: 0,
              customerId: 1,
              apdAppletInfo: Immutable.Map {
                  pinpad_data_entry_timeout: 20,
                  avs_zip: false,
                  avs_full_addr: true,
                  write_to_log: "Yes",
                  merchant_user: "AMS34VbVMSCRealDivId",
                  device_id: 999888,
                  merchant_descriptor: "ActiveNetTrainer",
                  pinpad_pin_entry_timeout: 40,
                  ams_timeout: 100,
                  terminal_location_state: "CA",
                  terminal_number: "007",
                  terminal_location_address: "937 Enterprise Dr",
                  usb_port_name: "COM5",
                  card_acceptor_name: "YMCA OF METROPOLITAN LOS ANGELES",
                  applet_name: "APDInterfaceApplet",
                  terminal_location_city: "Sacramento",
                  merchant_password: "AMS34VbVMSCRealDivId",
                  ams_url: "mstest.active.com",
                  device_type_id: 1,
                  pinpad_swipe_card_timeout: 50,
                  applets: "codebase=/device-libs/",
                  pinpad_device_timeout: 300,
                  ams_port: 443,
                  terminal_location_zip: "95825",
              },
              errorMessage: "",
              apdPaymentInfo: Immutable.Map {
                  receiptNumber: 1000778.005,
                  hasImmediatePaymentDue: false,
                  orderId: "Receipt ID 22916",
                  receiptPaymentId: 20423,
                  rno: 1,
                  committedReceiptId: 22916,
                  canCancelApdPayment: true,
                  isPayerCashCustomer: false,
                  orderDescriptor: "YMCA OF METROPOLITAN LOS ANGELES",
                  key: 1,
              },
          },
            "pinpadFail": Immutable.Map {
              shown: false,
          },
            "pinpadModal": Immutable.Map {
              title: "Processing Transaction",
              shown: true,
              apdInterfaceApplet: Immutable.Map {
              },
          },
          }
        }
      />
      <div
        id="pinpad-applet-container"
      />
    </div>
  </Modal>
</PinpadModal>
`;

exports[`index/Payment/components/Modals/PinPad/components/PinpadModal component should render correctly when hasn\`t shown modal 1`] = `
<PinpadModal
  addError={[Function]}
  clearError={[Function]}
  communicatingWithDevice={[Function]}
  getAPDServerStatus={[Function]}
  pinpad={
    Object {
      "payment": Immutable.Map {
        receiptHeaderId: 1234,
        agentId: 0,
        isRefund: false,
        currentPayIndex: 0,
        promptMessage: "",
        isNewCard: false,
        cardHolderInfo: Immutable.Map {
            cardholderZip: null,
            cardholderAddr: null,
        },
        pays: Immutable.List [
            Immutable.Map {
                ams_account_id: "APD AccountID",
                payment_type_id: 4,
                hasError: true,
                index: 0,
                payName: "Debit Card",
                amount: "370.00",
                apd_input_type: 1,
                paymentTypeId: 4,
                errorMessage: "unknown error",
            },
        ],
        communicating: false,
        debitCardId: 4,
        companyId: 0,
        customerId: 1,
        apdAppletInfo: Immutable.Map {
            pinpad_data_entry_timeout: 20,
            avs_zip: false,
            avs_full_addr: true,
            write_to_log: "Yes",
            merchant_user: "AMS34VbVMSCRealDivId",
            device_id: 999888,
            merchant_descriptor: "ActiveNetTrainer",
            pinpad_pin_entry_timeout: 40,
            ams_timeout: 100,
            terminal_location_state: "CA",
            terminal_number: "007",
            terminal_location_address: "937 Enterprise Dr",
            usb_port_name: "COM5",
            card_acceptor_name: "YMCA OF METROPOLITAN LOS ANGELES",
            applet_name: "APDInterfaceApplet",
            terminal_location_city: "Sacramento",
            merchant_password: "AMS34VbVMSCRealDivId",
            ams_url: "mstest.active.com",
            device_type_id: 1,
            pinpad_swipe_card_timeout: 50,
            applets: "codebase=/device-libs/",
            pinpad_device_timeout: 300,
            ams_port: 443,
            terminal_location_zip: "95825",
        },
        errorMessage: "",
        apdPaymentInfo: Immutable.Map {
            receiptNumber: 1000778.005,
            hasImmediatePaymentDue: false,
            orderId: "Receipt ID 22916",
            receiptPaymentId: 20423,
            rno: 1,
            committedReceiptId: 22916,
            canCancelApdPayment: true,
            isPayerCashCustomer: false,
            orderDescriptor: "YMCA OF METROPOLITAN LOS ANGELES",
            key: 1,
        },
    },
      "pinpadFail": Immutable.Map {
        shown: false,
    },
      "pinpadModal": Immutable.Map {
        title: "Processing Transaction",
        shown: false,
        apdInterfaceApplet: Immutable.Map {
        },
    },
    }
  }
  processAPDResponse={[Function]}
  processRejectAction={[Function]}
  processResolveAction={[Function]}
  setApdInterfaceApplet={[Function]}
  showFailActionBtns={[Function]}
  showModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  updateCurrentPayIndex={[Function]}
  updateErrorMessage={[Function]}
  updateModalTitle={[Function]}
  updatePromptMessage={[Function]}
  updateSuccessMessage={[Function]}
>
  <Modal
    className="pinpad-modal"
    shown={false}
    title="Processing Transaction"
  >
    <div
      className="modal-body"
    >
      <div
        id="pinpad-applet-container"
      />
    </div>
  </Modal>
</PinpadModal>
`;
